src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -2148,5 +2148,97 @@ namespace UiaOperationAbstractionTests
         {
             ElementAsVariantTest(true);
         }
+
+        // Tests the try block and catch block implementation both locally and remote. When an exception is encountered, the execution
+        // continues till the end of instructions without returning from the exception location, also records the exception
+        // in catch block.
+        void TryAndCatchBlockTest(const bool useRemoteOperations)
+        {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+            UiaElement element = calc;
+
+            auto guard = InitializeUiaOperationAbstraction(useRemoteOperations);
+
+            auto scope = UiaOperationScope::StartNew();
+
+            scope.BindInput(element);
+
+            UiaInt failureCode = 0;
+            UiaArray<UiaInt> numbers;
+            numbers.Append(1);
+            numbers.Append(2);
+            numbers.Append(3);
+            scope.TryCatch([&]()
+                {
+                    auto number = numbers.GetAt(3);
+                },
+                [&](UiaFailure failure)
```

Comment:
I'm actually a little surprised this still builds with all the constructors marked as deleted. I'll think a bit more about it to make sure we didn't miss something.

Reply:
The reason this builds is because of [copy elision guarantees in C++17](https://en.cppreference.com/w/cpp/language/copy_elision). Neat!